### PR TITLE
Ability to remove link preview from message

### PR DESCRIPTION
**Pull request by ratik21**
image

After removing:

image

### CODE REVIEW

File: src/components/message/index.tsx [hunk 1]
```diff
@@ -248,11 +254,20 @@ export class Message extends React.Component<Properties, State> {
                     </div>
                   )}
                   {message && this.renderMessageWithLinks()}
-                  {preview && (
-                    <LinkPreview
-                      url={preview.url}
-                      {...preview}
-                    />
+                  {preview && hidePreview === false && (
```
Comment (hkhanouchi) [inline src/components/message/index.tsx:257]: `{preview && !hidePreview && (`

File: src/store/messages/api.ts [hunk 2]
```diff
@@ -46,10 +46,11 @@ export async function editMessageApi(
   channelId: string,
   messageId: number,
   message: string,
-  mentionedUserIds: string[]
+  mentionedUserIds: string[],
+  data?: object
```
Comment (mrval5) [inline src/store/messages/api.ts:50]: what is data ? what is the structure of it that we are using in zOS ?

File: src/store/messages/saga.ts [hunk 3]
```diff
@@ -200,13 +201,14 @@ export function* deleteMessage(action) {
...
-  const { channelId, messageId, message, mentionedUserIds } = action.payload;
+  const { channelId, messageId, message, mentionedUserIds, data } = action.payload;
+
   const selectedMessage = yield select(messageSelector(messageId));
   const existingMessages = yield select(rawMessagesSelector(channelId));
 
   const messages = existingMessages.map((id) => {
     if (messageId === id) {
-      return { ...selectedMessage, updatedAt: Date.now(), message };
+      return { ...selectedMessage, updatedAt: Date.now(), message, hidePreview: data?.hidePreview ?? false };
```
Comment (mrval5) [inline src/store/messages/saga.ts:211]: Maybe:
```suggestion
      return { ...selectedMessage, updatedAt: Date.now(), message, hidePreview: Boolean(data?.hidePreview)};
```

File: src/store/messages/api.ts (hunk 2 above)
Comment (mrval5) [reply src/store/messages/api.ts:50]: I understood it's like optional `options` which it contains `hidePreview`, le's add the type please

**Review by mrval5 — CHANGES_REQUESTED** [review]

**Review by mrval5 — APPROVED** [review]

**Pull request merged**